web/packages/teleport/src/services/resources/resource.ts

```diff
@@ -60,7 +60,13 @@ class ResourceService {
             KeysEnum.UNIFIED_RESOURCES_NOT_SUPPORTED,
             'true'
           );
-          history.replace(cfg.getNodesRoute(clusterId));
+
+          // We do not want to push to a new path if they are requesting
+          // resoruces for an access request. We keep them on the page and instead
+          // show the old tables for requesting.
+          if (params.searchAsRoles !== 'yes') {
+            history.replace(cfg.getNodesRoute(clusterId));
+          }
```

Comment:
Since we have the same redirect in `teleport/UnifiedResources` we can remove it from here, no?